packages/pay/src/controllers/index.ts

```diff
@@ -82,6 +81,56 @@ export interface ControllerMap
   get<T extends StreamController>(key: new (...args: any[]) => T): T
 }
 
+export class StreamFulfill implements StreamReply {
+  readonly log: Logger
+  readonly frames?: Frame[]
+  readonly destinationAmount?: Integer
+
+  constructor(log: Logger, frames?: Frame[], destinationAmount?: Integer) {
+    this.log = log
+    this.frames = frames
+    this.destinationAmount = destinationAmount
+  }
+
+  isAuthentic() {
+    return true
+  }
+
+  isReject(): this is StreamReject {
```

Comment:
What syntax is that return type? Never seen it before

Reply:
`x is y` is a type guard, which is a function that returns a boolean to narrow the type of some variable. `x` refers to a variable (either the name of one of the parameters to that function, or `this` for an instance) and `y` is a type that `x` _could_ be. If the function/method returns true, TypeScript will treat the variable `x` as the more specific type `y`.

For example, if I have a `StreamReply` and ensure `isReject()` is true, that allows me to safely access the `ilpReject` property because TypeScript knows it exists.